plugins/output/adapt/importsourcecheck.js

```diff
@@ -0,0 +1,244 @@
+                        logger.log('info', files.file.path)
+                        helpers.unzip(files.file.path, COURSE_ROOT_FOLDER, function(error) {
+                            if(error) return cb2(error);
+                            importInfo['cleanupDirs'] = [files.file.path, COURSE_ROOT_FOLDER];
+                            cb2();
+                        });
+                    });
+                });
+            }
+        ], cb);
+    }
+
+    function findLanguages(cb) {
+        var courseLangs = [];
+        fs.readdir(COURSE_JSON_PATH, function (error, files) {
+            if (error) {
+                return cb(new Error(app.polyglot.t('app.importinvalidpackage')));
+            }
+            files.map(function (file) {
+                return path.join(COURSE_JSON_PATH, file);
+            }).filter(function (file) {
+                return fs.statSync(file).isDirectory();
+            }).forEach(function (file) {
+                courseLangs.push(path.basename(file));
+            });
+            COURSE_LANG = courseLangs[0] ? courseLangs[0] : 'en';
+            importInfo['COURSE_LANG'] = COURSE_LANG;
+            cb();
+        });
+    }
+
+    /**
+     * Checks course for any potential incompatibilities
+     */
+    function validateCoursePackage(done) {
+        // - Check framework version compatibility
+        // - Check we have all relevant json files using contentMap
+        // - Check assets
+        // - Check plugin versions
+        async.auto({
+            checkFramework: function(cb) {
+                fs.readJson(path.join(COURSE_ROOT_FOLDER, 'package.json'), function(error, versionJson) {
+                    if(error) return cb(error);
+                    helpers.checkFrameworkVersion(versionJson, function(error, data) {
+                        if(error) return cb(error);
+                        details.frameworkVersions = data;
+                        cb();
+                    });
+                });
+            },
+            checkContentJson: ['checkFramework', function(results, cb) {
+                async.eachSeries(Object.keys(contentMap), function(type, cb2) {
+                    var jsonPath = path.join(COURSE_JSON_PATH, (type !== 'config') ? COURSE_LANG : '', `${contentMap[type] || type}.json`);
+                    if (!fs.existsSync(jsonPath)) {
+                        return cb2(error);
+                    }
+                    cb2();
+                }, cb);
+            }],
+            checkAssetFolders: ['checkContentJson', function(results, cb) {
+                if (!cleanFormAssetDirs.length) {
+                    importInfo['assetFolders'] = Constants.Folders.ImportAssets;
+                    return cb();
+                }
+                var assetFolderError = false;
+                var missingFolders = [];
+                importInfo['assetFolders'] = cleanFormAssetDirs;
+                for (index = 0; index < importInfo['assetFolders'].length; ++index) {
```

Comment:
```suggestion
                for (var i = 0, j = importInfo['assetFolders'].length; i < j; ++i) {
```
(or similar)

Reply:
Need to switch the references from `index` to `i` inside the loop. Could do with a variable e.g.
```js
var assetFolder = importInfo.assetFolders[i];
```